gplearn/genetic.py

```diff
@@ -471,7 +471,10 @@ def fit(self, X, y, sample_weight=None):
 
         if isinstance(self, RegressorMixin):
             # Find the best individual in the final generation
-            self._program = self._programs[-1][np.argmin(fitness)]
+            if self._metric.greater_is_better:
+                self._program = self._programs[-1][np.argmax(fitness)]
+            else:
+                self._program = self._programs[-1][np.argmin(fitness)]
 
         if isinstance(self, TransformerMixin):
             # Find the best individuals in the final generation
```

Comment:
Presumably all the existing tests should pass with this change, just add an additional regression test to keep this bug removed in the future as well.